app/services/fuzzySearchService.js

```diff
@@ -0,0 +1,59 @@
+const Fuse = require('fuse.js')
+const { getAllPaymentData } = require('../services/databaseService')
+
+// Locally cached data
+let localCachedData = null
+async function getCachedPaymentData () {
+  if (!localCachedData) {
+    localCachedData = await getAllPaymentData()
+  }
+  return localCachedData
+}
+
+// search configuration
+const fuseSearchOptions = {
+  includeScore: true,
+  threshold: 0.3,
+  ignoreLocation: true,
+  useExtendedSearch: true,
+  keys: ['payee_name', 'part_postcode', 'town', 'county_council']
+}
+
+// Sort the records
+function getSortedValue (records, field) {
```

Comment:
I think getSortedValue can be simplified further by:
```suggestion
// define keys outside the function and use it also with fuseSearchOptions
const keys = ['payee_name', 'part_postcode', 'town', 'county_council']
function getSortedValue (records, field) {
   if(keys.includes(field)) {
     return records.sort((r1, r2) => r1[field]> r2[field] ? 1 : -1)
   }
   return records
}
```

Let me know what you think?

Reply:
agreed and updated the code to reflect the comments